packages/model-viewer/src/three-components/CachingGLTFLoader.ts

```diff
@@ -189,6 +199,9 @@ export class CachingGLTFLoader<T extends GLTFInstanceConstructor =
   async preload(
       url: string, element: ModelViewerElementBase,
       progressCallback: ProgressCallback = () => {}) {
+    if (withCredentials === true) {
+      this[$loader].setWithCredentials(withCredentials);
```

Comment:
Instead of setting this every time, how about you just set it inside of `setWithCredentials()`, and then you can also remove `withCredentials`. I don't think the getter is used anywhere anyway, right?

Reply:
I already tried but I have a typescript error :

```
src/three-components/CachingGLTFLoader.ts:130:5 - error TS7053: Element implicitly has an 'any' type because expression of type 'unique symbol' can't be used to index type 'typeof CachingGLTFLoader'.
  Property '["./CachingGLTFLoader.js".$loader]' does not exist on type 'typeof CachingGLTFLoader'.

130     this[$loader].setWithCredentials(state)
        ~~~~~~~~~~~~~
```

I'm do not learned typescript yet, if you have a workaround or an advice to fixe that, do not hesitate !

Reply:
Or maybe I do not understand correctly your proposal. I integrated 

```js
    if (withCredentials === true) {
      this[$loader].setWithCredentials(withCredentials);
    }
```

In the 

```js
  static setWithCredentials(state: boolean) {
    this[$loader].setWithCredentials(state)
  }
```

Reply:
Oh, I think the problem is `[$loader]` is a class member, but you made `setWithCredentials()` a static member, which can't access class members. Just make `setWithCredentials` a class member and it should be fine. We only create one shared CachingGLTFLoader object anyway, so all methods are effectively static anyway. `setDRACODecoderLocation` and such are static because they are called by a static public API, which yours is not.  That's just tricky Lit-element stuff.

Reply:
Understood, thanks !

So I moved the static setWithCredentials(state:boolean) inside the class. But now, in the `loading.ts` file, I can't access to the `setWithCredentials` :

```
src/features/loading.ts:430:27 - error TS2339: Property 'setWithCredentials' does not exist on type 'typeof CachingGLTFLoader'.

430         CachingGLTFLoader.setWithCredentials(this.withCredentials)
                              ~~~~~~~~~~~~~~~~~~
```

Reply:
Here is a new proposal, based on variable instead of functions.
Let me know what do you think about that. :)